src/utils/gauges.ts

```diff
@@ -372,3 +430,171 @@ export function areGaugesActive(chainId?: ChainId): boolean {
     IS_VESDL_LIVE
   )
 }
+
+async function getGaugeCount(
+  childGaugeFactory: ChildGaugeFactory | null,
+  gaugeController: GaugeController | null,
+): Promise<BigNumber> {
+  if (childGaugeFactory) {
+    return await childGaugeFactory.get_gauge_count()
+  }
+
+  if (gaugeController) {
+    return await gaugeController.n_gauges()
+  }
+
+  return Promise.resolve(Zero)
+}
+
+function retrieveGaugeAddresses(
+  gaugeControllerMultiCall: MulticallContract<GaugeController>,
+  childGaugeFactoryMultiCall: MulticallContract<ChildGaugeFactory>,
+): (
+  value: number,
+  index: number,
+  array: number[],
+) => MulticallCall<
+  [arg0: BigNumberish, overrides?: CallOverrides | undefined],
+  string
+> {
+  if (gaugeControllerMultiCall.address) {
+    return (value) => gaugeControllerMultiCall.gauges(value)
+  }
+  return (value) => childGaugeFactoryMultiCall.get_gauge(value)
+}
+
+const retrieveGaugeContracts = (
+  gaugeAddresses: string[],
+  chainId: ChainId,
+): MulticallContract<LiquidityGaugeV5>[] | MulticallContract<ChildGauge>[] => {
+  if (chainId === ChainId.MAINNET) {
+    return gaugeAddresses.map((address) =>
+      createMultiCallContract<LiquidityGaugeV5>(
+        address,
+        LIQUIDITY_GAUGE_V5_ABI,
+      ),
+    )
+  }
+
+  return gaugeAddresses.map((address) =>
+    createMultiCallContract<ChildGauge>(address, CHILD_GAUGE_ABI),
+  )
+}
+
+async function getSDLRates(
+  gaugeMinterContract: Minter | null,
+  gaugeRelativeWeights: BigNumber[],
+  gaugeMulticallContracts: MulticallContract<ChildGauge>[],
+  ethCallProvider: MulticallProvider,
+) {
+  if (gaugeMinterContract) {
+    const minterSDLRate = await gaugeMinterContract.rate()
+    return gaugeRelativeWeights.map((weight) =>
+      minterSDLRate.mul(weight).div(BN_1E18),
+    )
+  }
+
+  const currentTimeStamp = Date.now() / 1000
+  return await ethCallProvider.tryAll(
+    gaugeMulticallContracts.map((contract) =>
+      contract.inflation_rate(Math.floor(currentTimeStamp / 604800)),
+    ),
+  )
+}
+
+async function getGaugeRewardCounts(
+  gaugeMulticallContracts:
+    | MulticallContract<LiquidityGaugeV5>[]
+    | MulticallContract<ChildGauge>[],
+  ethCallProvider: MulticallProvider,
+) {
+  return (
+    await ethCallProvider.tryAll(
+      gaugeMulticallContracts.map((contract) => contract.reward_count()),
+    )
+  ).map((count) => count || Zero)
+}
+
+async function getGaugeRewardTokens(
+  gaugeMulticallContracts: (
+    | MulticallContract<ChildGauge>
+    | MulticallContract<LiquidityGaugeV5>
+  )[],
+  gaugeRewardCounts: BigNumber[],
+  ethCallProvider: MulticallProvider,
+) {
+  return Promise.all(
+    gaugeRewardCounts.map((count, index) =>
+      ethCallProvider.all(
```

Comment:
Should this be tryAll?

Reply:
Good catch. Will update.

Reply:
At some point we should start using `chunkedTryAll` everywhere to avoid the case where we are trying to make more many calls in one operation than multicall can handle. See `useBasicTokens` for example